pkg/providers/amifamily/resolver.go

```diff
@@ -134,25 +136,43 @@ func (r DefaultResolver) Resolve(nodeClass *v1.EC2NodeClass, nodeClaim *karpv1.N
 		// In order to support reserved ENIs for CNI custom networking setups,
 		// we need to pass down the max-pods calculation to the kubelet.
 		// This requires that we resolve a unique launch template per max-pods value.
-		// Similarly, instance types configured with EfAs require unique launch templates depending on the number of
+		// Similarly, instance types configured with EFAs require unique launch templates depending on the number of
 		// EFAs they support.
+		// Reservations IDs are also included since we need to create a separate LaunchTemplate per reservation ID when
+		// launching reserved capacity. If it's a reserved capacity launch, we've already filtered the instance types
+		// further up the call stack.
 		type launchTemplateParams struct {
-			efaCount int
-			maxPods  int
+			efaCount       int
+			maxPods        int
+			reservationIDs string
 		}
-		paramsToInstanceTypes := lo.GroupBy(instanceTypes, func(instanceType *cloudprovider.InstanceType) launchTemplateParams {
+		paramsToInstanceTypes := lo.GroupBy(instanceTypes, func(it *cloudprovider.InstanceType) launchTemplateParams {
+			var reservationIDs []string
+			if capacityType == karpv1.CapacityTypeReserved {
+				for i := range it.Offerings {
+					if it.Offerings[i].Requirements.Get(karpv1.CapacityTypeLabelKey).Any() != karpv1.CapacityTypeReserved {
+						continue
+					}
+					reservationIDs = append(reservationIDs, it.Offerings[i].Requirements.Get(v1.LabelCapacityReservationID).Any())
+				}
+			}
 			return launchTemplateParams{
 				efaCount: lo.Ternary(
 					lo.Contains(lo.Keys(nodeClaim.Spec.Resources.Requests), v1.ResourceEFA),
-					int(lo.ToPtr(instanceType.Capacity[v1.ResourceEFA]).Value()),
+					int(lo.ToPtr(it.Capacity[v1.ResourceEFA]).Value()),
 					0,
 				),
-				maxPods: int(instanceType.Capacity.Pods().Value()),
+				maxPods: int(it.Capacity.Pods().Value()),
+				// If we're dealing with reserved instances, there's only going to be a single instance per group. This invariant
+				// is due to reservation IDs not being shared across instance types. Because of this, we don't need to worry about
+				// ordering in this string.
+				reservationIDs: strings.Join(reservationIDs, ","),
```

Comment:
How many launch templates are we going to produce here in the worst-case? Is it possible that we create so many launch templates that we continually get rate-limited? Should we truncate this list down based on the number of reservations that we get passed-through so that we don't run this risk?

Reply:
At the end of the day, sending more capacity reservations to fleet is just going to give us a higher chance that we actually launch something that is available so reducing this number down (to me) seems fine

Reply:
Unfortunately, it ends up being 1 launch template per capacity reservation. I wasn't too concerned about the total number though since launches with reserved instances are naturally going to have fewer total instance types, and the number of reservations is directly managed by the user. 

> At the end of the day, sending more capacity reservations to fleet is just going to give us a higher chance that we actually launch something that is available so reducing this number down (to me) seems fine

I don't think I follow this, reducing this number down would also reduce the number of capacity reservations we send to fleet. Could you elaborate?

Reply:
My point was basically that if we truncated instance types down we would create less launch templates

Reply:
Yeah, that's kind of where I'm at. I'd like to provide maximum flexibility, at least to start. If we knew we were the sole consumer, I'd be happy to limit to 1.